src/api/controllers/UserController.ts

```diff
@@ -49,38 +49,30 @@ export class UserController {
   }
 
   @Get('postId/:id/')
-    async getUserByPostId(@Params() params: UuidParam): Promise<GetUserResponse | ErrorResponse> {
-      try {
-        return { success: true, user: await this.userService.getUserByPostId(params.id) }; 
-      } catch (error) {
-        return { success: false, error: getErrorMessage(error) }
-      }
-    }
-
-  @Post('email/')
-  async getUserByEmail(@Body() getUserByEmailRequest: GetUserByEmailRequest):
-      Promise<GetUserResponse | ErrorResponse> {
+  async getUserByPostId(@Params() params: UuidParam): Promise<GetUserResponse | ErrorResponse> {
     try {
-      return { success: true, user: await this.userService.getUserByEmail(getUserByEmailRequest.email) };
+      return { success: true, user: await this.userService.getUserByPostId(params.id) }; 
     } catch (error) {
       return { success: false, error: getErrorMessage(error) }
     }
   }
-  
-  @Post()
-  async createUser(@Body() createUserRequest: CreateUserRequest): Promise<GetUserResponse | ErrorResponse> {
+
+  @Get('save/userId/:userId/postId/:postId/') 
+  async savePost(@Params() params: PostAndUserUuidParam): Promise<GetPostResponse | ErrorResponse> {
     try {
-      const user = await this.userService.createUser(createUserRequest);
-      return { success: true, user: user };
+      const post = await this.userService.savePost(params.userId, params.postId)
+      return { success: true, post }; 
```

Comment:
in a different PR please undo all the success, data paradigm responses - you should just be returning the post and success should be determined on whether or not the status code is < 400

Reply:
Shungo is working on it in a different branch